docs/run-keep-node.adoc

```diff
@@ -0,0 +1,359 @@
+=== Configuration
+
+The client expects configuration options to be specified in a <<config-file,config file>> 
+or passed as <<cli,CLI flags>>. If you specify an option by using a parameter on 
+the command line, it will override the value read from the configuration file.
+
+[#ethereum-key]
+==== Operator's Ethereum Key File
+
+Ethereum Key File is expected to be encrypted with a password. The password has
+to be provided in prompt after the client starts or configured as a `KEEP_ETHEREUM_PASSWORD`
+environment variable `KEEP_ETHEREUM_PASSWORD`.
+
+[#ethereum-api]
+==== Ethereum API
+
+A Keep Node requires a connection to a WebSocket Ethereum API. You should obtain
+a WS API URL from a service provider 
+(e.g. link:https://www.alchemy.com/[Alchemy], link:https://www.infura.io/[Infura]) 
+or run your own Ethereum node
+(e.g. link:https://geth.ethereum.org/[Geth]).
+
+[#cli]
+==== CLI Options
+
+[source,bash,role=small]
+----
+include::./resources/client-start-help[]
+----
+
+[#config-file]
+==== Config File
+
+Application configuration can be stored in a file and passed to the application
+with the `--config` flag. 
+
+Example:
+[source,bash]
+----
+./keep-client --config /path/to/your/config.toml start
+----
+
+Configuration files in formats TOML, YAML and JSON are supported.
+
+Sample configuration file:
+
+[source,toml,role=small]
+----
+include::../configs/config.toml.SAMPLE[]
+----
+
+[#storage]
+==== Persistance
+
+A client will produce and store data on disk. The directory should be provided to
+the client under `storage.DataDir` (flag: `--storage.dataDir`) configuration property.
+
+IMPORTANT: It is crucial to ensure the data directory is persisted and backed up on
+a regular basis.
+
+==== Network
+
+Default port mappings.
+
+[%header,cols=2*]
+|===
+|Egress
+|Port
+
+|Keep Network
+| TCP: `3919`
+|===
+
+[%header,cols=2*]
+|===
+|Ingress
+|Port
+
+|Keep Network
+|`3919`
+|===
+
+If you set a different `port` in your configuration, or configure `peers` with
+non-default ports configured, firewall rules will need to be adjusted accordingly.
+
+==== Minimum Required Configuration
+
+The minimum required configuration for the client to start covers setting:
+
+- `KEEP_ETHEREUM_PASSWORD` environment variable (see: <<ethereum-key>> section),
+- `ethereum.url` config property (see: <<ethereum-api>> section),
+- `ethereum.keyFile` config property (see: <<ethereum-key>> section),
+- `storage.dataDir` config property (see: <<storage>> section).
+
+=== Installation
+
+// TODO: Add binaries once release process is ready.
+// ==== Binaries
+
+// TODO: Update with mainnet Docker image details
+// ==== Docker
+// [source,shell,role=small]
+// ----
+// include::resources/docker-start-mainnet-sample[]
+// ----
+
+==== Build from Source
+
+See our xref:development/README.adoc#build-client[developer docs].
+
+=== Run
+
+Sample command to run the Keep Client:
+
+[source,shell,role=small]
+----
+include::resources/client-start-mainnet-sample[]
+----
```

Comment:
Should we provide Docker instructions here as well?

Reply:
887c3f0e692be378a61f3863a2391bf7c42f5328